Capri.Services/Proposals/ProposalFilter.cs

```diff
@@ -0,0 +1,165 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Capri.Database.Entities;
+using Capri.Web.ViewModels.Proposal;
+
+namespace Capri.Services.Proposals
+{
+    public class ProposalFilter : IProposalFilter
```

Comment:
At first i think it shouldn't be in separate directory. Just think that it's only GetProposal thing with just a few changes, but domain is still the same "returning things from database".

Other thing is writing something like this from scratch it's not a good idea. Just check if there is no ready to use components out of the box. 

After 2min research I've found:
https://github.com/Biarity/Sieve

Reply:
Also when your making multiple linq queries on object to do filtering don't use IEnumerable. It will load all database to memory due to query materialization. If you want to use multiple queries, you have to use IQueryable. It won't materialize query

Reply:
> At first i think it shouldn't be in separate directory. Just think that it's only GetProposal thing with just a few changes, but domain is still the same "returning things from database".

But we were talking about separation of the services, because we have some mess there

Reply:
Ok so I will ask you: what's difference in domain of that functionalities?